src/NuGet.Core/NuGet.Common/Telemetry/TelemetryEvent.cs

```diff
@@ -12,6 +13,9 @@ public class TelemetryEvent
     {
         private IDictionary<string, object> _properties;
         private IDictionary<string, object> _piiProperties;
+        private IDictionary<string, IEnumerable<string>> _piiLists;
+        private IDictionary<string, Tuple<string, string>> _piiPackages;
+        private IDictionary<string, IEnumerable<Tuple<string, string>>> _piiPackageLists;
```

Comment:
I'm not a big fan of this for several reasons. Firstly, without looking where these last two are used, it's impossible to guess what the `Tuple`'s `Item1` and `Item2` represent. I'm ok with `Tuple` being used in the same method it's declared. With C#7, ValueTuple with named parameters are a far more readable option, but creating a class or struct is also preferable to me than the old `Tuple`. Secondly, I don't like adding so many specific data types to a generic class. If more and more telemetry types get added, it will be unwieldy to have more and more lists added. I'm obviously biased since I wrote it, but I think [my PR's recursive approach](https://github.com/NuGet/NuGet.Client/pull/3105/files#diff-6ab855ae420f5f3e602505e91556859bR49-R94) is better.

However, since I'm going to merge my PR soon, I can refactor this code during merge conflict resolution, so this comment isn't a request for change, more of an FYI.